test/src/node.rs

```diff
@@ -54,13 +54,18 @@ impl Node {
         binary: &str,
         p2p_port: u16,
         rpc_port: u16,
-        case_name: &str,
+        case_working_dir: PathBuf,
         node_index: &str,
     ) -> Self {
         let rpc_client = RpcClient::new(&format!("http://127.0.0.1:{}/", rpc_port));
+        let working_dir = case_working_dir
+            .join(node_index)
+            .as_path()
+            .display()
+            .to_string();
```

Comment:
unnecessary `as_path`

```suggestion
        let working_dir = case_working_dir
            .join(node_index)
            .display()
            .to_string();
```